build/Dockerfile

```diff
@@ -8,26 +18,28 @@ COPY go.mod go.mod
 COPY go.sum go.sum
 # cache deps before building and copying source so that we don't need to re-download as much
 # and so that source changes don't invalidate our downloaded layer
-RUN go mod download
+# In downstream this needs to be a cached vendor/ folder, so do same upstream
+RUN go mod tidy; go mod vendor
```

Comment:
I don't think we should be vendoring in the dockerfile at all. If we're already swapping out lines (`go mod vendor` vs copying in a tarball) we may as well swap in a `-mod vendor` to the dockerfile when its needed. Otherwise we're invalidating the build cache and required a full vendor download whenever the source code is changed.

Reply:
> required a full vendor download whenever the source code is changed.

Hey, you just described a Brew build! :P 

But seriously... I've figured out what I need for downstream, and it's this:

1. bootstrap the vendor folder:

```
RUN go mod tidy; go mod vendor
COPY . .
RUN export ARCH="$(uname -m)" && if [[ ${ARCH} == "x86_64" ]]; then export ARCH="amd64"; elif [[ ${ARCH} == "aarch64" ]]; then export ARCH="arm64"; fi && \
  go mod tidy || true; go mod vendor || true; CGO_ENABLED=0 GOOS=linux GOARCH=${ARCH} GO111MODULE=on go build -mod=vendor
```

2. then commit the vendor folder

3. rebuild like this in Brew/offline:
```
COPY . .
RUN export ARCH="$(uname -m)" && if [[ ${ARCH} == "x86_64" ]]; then export ARCH="amd64"; elif [[ ${ARCH} == "aarch64" ]]; then export ARCH="arm64"; fi && \
  CGO_ENABLED=0 GOOS=linux GOARCH=${ARCH} GO111MODULE=on go build -mod=vendor
```

If there's a better way, please let me know. If you don't like my changes being applied in upstream I can just convert upstream into something that works for bootstrap and brew builds.